### ISSUE TITLE
Error occured during  Ensure Apache is installed on RHEL task.

### ISSUE THREAD

**Issue by TahaMateen** [geerlingguy/ansible-role-apache#155]
while using the role in my playbook i am getting the below error:

 (item=[u'httpd24', u'httpd24-devel', u'mod24_ssl', u'openssh']) => {"changed": false, "failed": true, "item": ["httpd24", "httpd24-devel", "mod24_ssl", "openssh"], "msg": "No package matching 'httpd24' found available, installed or updated", "rc": 126, "results": ["No package matching 'httpd24' found available, installed or updated"]}

Kindly help me out on this error.

**Comment by z0ph** [comment]
Hi there, 

Same issue here, I'm on Amazon Linux, I've used : 

```
# Set Distribution : Amazon Linux
ansible_distribution: "Amazon"
ansible_distribution_major_version: "NA"
```


```
TASK [geerlingguy.apache : Ensure Apache is installed on RHEL.] *****************************************************************************************************************************************
failed: [xxx-xxx] (item=[u'httpd24', u'httpd24-devel', u'mod24_ssl', u'openssh']) => {"changed": false, "item": ["httpd24", "httpd24-devel", "mod24_ssl", "openssh"], "msg": "Error: httpd24-tools conflicts with httpd-tools-2.2.34-1.16.amzn1.x86_64\nError: httpd24 conflicts with httpd-2.2.34-1.16.amzn1.x86_64\n", "rc": 1, "results": ["openssh-7.4p1-16.71.amzn1.x86_64 providing openssh is already installed", "Loaded plugins: priorities, update-motd, upgrade-helper\nResolving Dependencies\n--> Running transaction check\n---> Package httpd24.x86_64 0:2.4.34-1.82.amzn1 will be installed\n--> Processing Dependency: httpd24-tools = 2.4.34-1.82.amzn1 for package: httpd24-2.4.34-1.82.amzn1.x86_64\n---> Package httpd24-devel.x86_64 0:2.4.34-1.82.amzn1 will be installed\n--> Processing Dependency: apr-util-devel for package: httpd24-devel-2.4.34-1.82.amzn1.x86_64\n--> Processing Dependency: apr-devel for package: httpd24-devel-2.4.34-1.82.amzn1.x86_64\n---> Package mod24_ssl.x86_64 1:2.4.34-1.82.amzn1 will be installed\n--> Running transaction check\n---> Package apr-devel.x86_64 0:1.5.2-5.13.amzn1 will be installed\n---> Package apr-util-devel.x86_64 0:1.5.4-6.18.amzn1 will be installed\n--> Processing Dependency: expat-devel(x86-64) >= 2.1.0 for package: apr-util-devel-1.5.4-6.18.amzn1.x86_64\n--> Processing Dependency: openldap-devel(x86-64) for package: apr-util-devel-1.5.4-6.18.amzn1.x86_64\n--> Processing Dependency: db4-devel(x86-64) for package: apr-util-devel-1.5.4-6.18.amzn1.x86_64\n---> Package httpd24-tools.x86_64 0:2.4.34-1.82.amzn1 will be installed\n--> Running transaction check\n---> Package db4-devel.x86_64 0:4.7.25-18.11.amzn1 will be installed\n--> Processing Dependency: db4-cxx = 4.7.25-18.11.amzn1 for package: db4-devel-4.7.25-18.11.amzn1.x86_64\n---> Package expat-devel.x86_64 0:2.1.0-10.21.amzn1 will be installed\n---> Package openldap-devel.x86_64 0:2.4.40-12.30.amzn1 will be installed\n--> Processing Dependency: cyrus-sasl-devel >= 2.1 for package: openldap-devel-2.4.40-12.30.amzn1.x86_64\n--> Running transaction check\n---> Package cyrus-sasl-devel.x86_64 0:2.1.23-13.16.amzn1 will be installed\n---> Package db4-cxx.x86_64 0:4.7.25-18.11.amzn1 will be installed\n--> Processing Conflict: httpd24-tools-2.4.34-1.82.amzn1.x86_64 conflicts httpd-tools < 2.4.34\n--> Processing Conflict: httpd24-2.4.34-1.82.amzn1.x86_64 conflicts httpd < 2.4.34\n--> Finished Dependency Resolution\n You could try using --skip-broken to work around the problem\n You could try running: rpm -Va --nofiles --nodigest\n"]}
        to retry, use: --limit @/home/ec2-user/main.retry
```

**Comment by smeek** [comment]
I see a similar issue:

`TASK [geerlingguy.apache : Ensure Apache is installed on RHEL.] ****************
fatal: [xxx.xxx.xxx.xxx]: FAILED! => {"changed": false, "msg": "The Python 2 yum module is needed for this module. If you require Python 3 support use the `dnf` Ansible module instead."}`

I'm using Vagrant 2.1.5 and Ansible 2.6.0 to spin-up a Fedora 28 VM with virtualbox and a `bento` box:
`bento/fedora-28    (virtualbox, 201808.24.0)`

Role version: `geerlingguy.apache (3.0.2)`

**Comment by smeek** [comment]
Interestingly, in my local testing, changing from `yum` to `package` in `tasks/setup-RedHat.yml` seems to work.
Slightly worried that `enablerepo` is not listed as a parameter for `package` though, but I am guessing it picks it up from either the `yum` or `dnf` modules somehow.

**Comment by smeek** [comment]
Created a [PR](https://github.com/geerlingguy/ansible-role-apache/pull/158) with this change...

**Comment by tgallacher** [comment]
I am also running into this error, and I see that the above PR has not yet been released (as of v3.0.3). 

@geerlingguy any ETA on when https://github.com/geerlingguy/ansible-role-apache/pull/158 might be released? 

thanks
